### ISSUE TITLE
Make proper executable with entry_point

### ISSUE THREAD

**Issue by bollwyvl** [martinRenou/json-lsp#2]
I'd recommend the `entry_point` approach, as it is generally preferable to using `python -m` if possible, as `-m` can have... unwanted side-effects. Further, a `conda` downstream will do a good job of making it robust on different platforms.

**Comment by martinRenou** [comment]
I did use the `entry_point` approach: https://github.com/martinRenou/json-lsp/blob/master/setup.py#L20-L24

Should I do it in another way?

**Comment by bollwyvl** [comment]
sorry, yes, the lsp entry_point is :+1: 

I mean for the thing-what-runs, e.g. https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

```py
setup(
    ...
    entry_points = {
        'console_scripts': ['json-lsp-or-whatever=json_lsp.main:main'],
    }
    ...
)
```

**Comment by martinRenou** [comment]
I see. So I would need two entry points, one for the console script (that actually runs the json-lsp binary) and one for lsp (that provides the spec)?

**Comment by bollwyvl** [comment]
Or go to `flit` 
https://flit.readthedocs.io/en/latest/pyproject_toml.html#scripts-section

...and get reproducible wheels, too :ferris_wheel:

**Comment by bollwyvl** [comment]
right. then you can either use the shell helper instead of the python one... or just return the dictionary, and not have a dependency on on `jupyter_lsp` at all... that was for us to reduce boilerplate. we did the thing for `pyls` for... reasons. probably windows. don't remember now.

**Comment by martinRenou** [comment]
I didn't know about flit! Thanks for the pointer
